packages/dds/tree/src/feature-libraries/modular-schema/modularChangeFamily.ts

```diff
@@ -2791,13 +2829,23 @@ function getRevInfoFromTaggedChanges(changes: TaggedChange<ModularChangeset>[]):
 	maxId: ChangesetLocalId;
 } {
 	let maxId = -1;
-	const revInfos: RevisionInfo[] = [];
+	const revInfoList: RevisionInfo[] = [];
 	for (const taggedChange of changes) {
 		const change = taggedChange.change;
 		maxId = Math.max(change.maxId ?? -1, maxId);
-		revInfos.push(...revisionInfoFromTaggedChange(taggedChange));
+		revInfoList.push(...revisionInfoFromTaggedChange(taggedChange));
 	}
 
+	// Remove duplicate revision infos. Each change can have a revision info and it may be the same as
+	// other changes in the list.
+	const revInfos = revInfoList.filter(
```

Comment:
Is this still required? How do we get into a situation where there a duplicates?

Reply:
Ah, I remember now, this is because we compose changes that have the same revision (problaby when a move is made).
Since the idea of composing multiple changes with the same revision is specific to the changeset building logic, I think it would make sense to address it there (in the ModularEditBuilder.buildChanges method). In that method, you should be able to prevent duplicates by populating a `Set<RevisionTag>` as you iterate over the revision list. It will be more effcient than using filter and findIndex.

Reply:
Updated it to happen in `buildChanges`. Please take a look.